### ISSUE TITLE
Split filter not working on an array of numbers

### ISSUE THREAD

**Issue by geoffroya** [logstash-plugins/logstash-filter-split#13]
Hello

I want to split a single event into several ones, based on an array.
My original event is as follows:

```
{
       "error" => false,
       "message" => "broker topic offsets returned",
       "offsets" => [
        [0] 140395,
        [1] 140395,
        [2] 140396,
        [3] 140396,
        [4] 140395
    ],
    "@version" => "1",
    "@timestamp" => 2016-04-07T06:57:20.963Z
}
```

Here is my filter config:

```
split {
    field => "offsets"
    target => "offset"
}
```

When processing the filter, Logstash stops with the following output:!

```
Exception in pipelineworker, the pipeline stopped processing new events, please check your filter configuration and restart Logstash.
{
    "exception"=>#<NoMethodError: undefined method `empty?' for 140395:Fixnum>,
    "backtrace"=>[
        "D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-filter-split-2.0.4/lib/logstash/filters/split.rb:57:in `filter'",
        "org/jruby/RubyArray.java:1613:in `each'",
        "D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-filter-split-2.0.4/lib/logstash/filters/split.rb:56:in `filter'",
        "D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/filters/base.rb:151:in `multi_filter'", 
        "org/jruby/RubyArray.java:1613:in `each'", 
        "D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/filters/base.rb:148:in `multi_filter'", 
        "(eval):45:in `filter_func'", 
        "D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/pipeline.rb:271:in `filter_batch'", 
        "org/jruby/RubyArray.java:1613:in `each'", "org/jruby/RubyEnumerable.java:852:in `inject'",
        "D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/pipeline.rb:269:in `filter_batch'",
        "D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/pipeline.rb:227:in `worker_loop'",
        "D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/pipeline.rb:205:in `start_workers'"],
    :level=>:error
}
NoMethodError: undefined method `empty?' for 140395:Fixnum
         filter at D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-filter-split-2.0.4/lib/logstash/filters/split.rb:57
           each at org/jruby/RubyArray.java:1613
         filter at D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-filter-split-2.0.4/lib/logstash/filters/split.rb:56
   multi_filter at D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/filters/base.rb:151
           each at org/jruby/RubyArray.java:1613
   multi_filter at D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/filters/base.rb:148
    filter_func at (eval):45
   filter_batch at D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/pipeline.rb:271
           each at org/jruby/RubyArray.java:1613
         inject at org/jruby/RubyEnumerable.java:852
   filter_batch at D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/pipeline.rb:269
    worker_loop at D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/pipeline.rb:227
  start_workers at D:/Applis/logstash-2.3.0/vendor/bundle/jruby/1.9/gems/logstash-core-2.3.0-java/lib/logstash/pipeline.rb:205
```

Seeing **undefined method `empty?' for 140395:Fixnum>** I added a mutate to convert my numbers to String, and the split did the job.

Furthermore, it would be nice if the split filter could add another field to track the _index_ of the split field.
In my case, It would be nice to have the split filter configured like:

```
filter {
  field => "offsets"
  target => "offset"
  split_index_field => "partition_id"
  remove_field => "offsets"
}
```

Thus resulting in events like this:

```
{
       "error" => false,
       "message" => "broker topic offsets returned",
       "offset" => 140395,
       "partition_id" => 0,
       "@version" => "1",
       "@timestamp" => 2016-04-07T06:57:20.963Z
}

{
       "error" => false,
       "message" => "broker topic offsets returned",
       "offset" => 140395,
       "partition_id" => 1,
       "@version" => "1",
       "@timestamp" => 2016-04-07T06:57:20.963Z
}

{
       "error" => false,
       "message" => "broker topic offsets returned",
       "offset" => 140396,
       "partition_id" => 2,
       "@version" => "1",
       "@timestamp" => 2016-04-07T06:57:20.963Z
}

and so on...
```

Thank you for your feedback.

Geoff

**Comment by colinsurprenant** [comment]
I confirm this is still a problem on filter 3.1.6 and the bug is https://github.com/logstash-plugins/logstash-filter-split/blob/fb040c751cfd882637f29070f68e36d948a29979/lib/logstash/filters/split.rb#L89 
where it assumes the split values are strings.

**Comment by colinsurprenant** [comment]
PR #36 submitted .

**Comment by colinsurprenant** [comment]
Fixed in v3.1.7
